mesonbuild/rewriter.py

```diff
@@ -646,18 +808,26 @@ def arg_list_from_node(self, n: BaseNode) -> T.List[BaseNode]:
             args = n.arguments
         return args
 
-    def rm_src_or_extra(self, op: str, target: IntrospectionBuildTarget, to_be_removed: T.List[str], to_sort_nodes: T.List[ArgumentNode]) -> None:
+    def rm_src_or_extra(self, op: str, target: IntrospectionBuildTarget, to_be_removed: T.List[str], to_sort_nodes: T.List[T.Union[FunctionNode, ArrayNode]]) -> None:
+        assert op in set(['src_rm', 'extra_files_rm'])
+        cwd = Path(os.getcwd())
+        source_root_abs = cwd / self.interpreter.source_root
+
         # Helper to find the exact string node and its parent
-        def find_node(src: str) -> T.Tuple[BaseNode, StringNode]:
+        def find_node(src: str) -> T.Tuple[T.Optional[BaseNode], T.Optional[StringNode]]:
             if op == 'src_rm':
-                nodes = target.source_nodes
+                nodes = self.interpreter.dataflow_dag.reachable(set(target.source_nodes), True).union(set([target.node]))
             elif op == 'extra_files_rm':
-                nodes = target.extra_files
+                nodes = self.interpreter.dataflow_dag.reachable(set([target.extra_files]), True)
```

Comment:
No need for intermediate list. You should use `{target.extra_files}`.

Reply:
Done